lib/rules/selector-anb-no-unmatchable/index.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const { parse } = require('css-tree');
+const {
+	aNPlusBNotationPseudoClasses,
+	aNPlusBOfSNotationPseudoClasses,
+} = require('../../reference/selectors');
+const isStandardSyntaxRule = require('../../utils/isStandardSyntaxRule');
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'selector-anb-no-unmatchable';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (pseudoClass) => `Unexpected unmatchable An+B selector "${pseudoClass}"`,
+	rejectedParseError: (selector) => `Cannot parse selector "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/selector-anb-no-unmatchable',
+};
+
+function isUnmatchableNth(/** @type {import('css-tree').AnPlusB} */ nth) {
+	const { a, b } = nth;
+
+	if (a !== null && a !== '0' && a !== '-0') {
+		return false;
+	}
+
+	if (b !== null && b !== '0' && b !== '-0') {
+		return false;
+	}
+
+	return true;
+}
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) {
+				return;
+			}
+
+			ruleNode.selectors.forEach((selector) => {
+				let cssTreeSelector;
+
+				try {
+					cssTreeSelector = parse(selector, { context: 'selector', positions: true });
+				} catch (e) {
+					report({
+						message: messages.rejectedParseError,
+						messageArgs: [selector],
+						node: ruleNode,
+						word: selector,
+						result,
+						ruleName,
+					});
+
```

Comment:
```suggestion
```

Let's silently catch parse errors, as it's out of the scope of this rule.

CSSTree is less forgiving than the PostCSS selector parser, and this will likely flag non-standard language extensions.